### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by mewmew** [Redcrafter/nesEmu#2]
Hi @Redcrafter!

I came across your NES emulator in the release notes of Dear ImGui. I think it is a really brilliant way to visualize all that is happening within the NES console. Thanks a lot for sharing this project.

I tried to compile nesEmu at rev 2ff31ff6c6dcb2022f518fac4d632c971d6fd8e6, but ran in to the following build error:

```
./nesEmu/src/Emulation/NES/Windows/disassembler.cpp: In function ‘int Nes::InstructionSize(Nes::AddressingModes)’:
./nesEmu/src/Emulation/NES/Windows/disassembler.cpp:226:46: error: no matching function for call to ‘std::exception::exception(const char [12])’
  226 |   default: throw std::exception("unreachable");
      |                                              ^
```

Clang and GCC versions below:

```
$ clang -v
clang version 11.0.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/10.2.0
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/10.2.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0
Found candidate GCC installation: /usr/lib64/gcc/x86_64-pc-linux-gnu/10.2.0
Selected GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/10.2.0
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64

$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --with-isl --with-linker-hash-style=gnu --with-system-zlib --enable-__cxa_atexit --enable-cet=auto --enable-checking=release --enable-clocale=gnu --enable-default-pie --enable-default-ssp --enable-gnu-indirect-function --enable-gnu-unique-object --enable-install-libiberty --enable-linker-build-id --enable-lto --enable-multilib --enable-plugin --enable-shared --enable-threads=posix --disable-libssp --disable-libstdcxx-pch --disable-libunwind-exceptions --disable-werror gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.0 (GCC) 
```

Cheers,
Robin

**Comment by mewmew** [comment]
I also ran into this build error. Perhaps I configured something incorrectly?

```
./nesEmu/src/main.cpp:32:10: fatal error: Emulation/GB/GameboyCore.h: No such file or directory
   32 | #include "Emulation/GB/GameboyCore.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I ran these commands to build the project:

```
git clone --recursive https://github.com/Redcrafter/nesEmu.git
mkdir build
cd build
cmake ../
make
```

**Comment by galaxyhaxz** [comment]
> I also ran into this build error. Perhaps I configured something incorrectly?

Readme says gameboy emulation is a WIP so likely a placeholder. That error should be fixable by removing that line :P

**Comment by mewmew** [comment]
> Readme says gameboy emulation is a WIP so likely a placeholder. That error should be fixable by removing that line :P

Yes. Worked well after removing that line ^^

**Comment by galaxyhaxz** [comment]
This popped up in my notifications after you starred it. As a die hard NES fan it's exciting to see. The 6502 processor was the very first thing I learned to code for. Hopefully this emulator will phase out the old FCEUX debugger with one that has proper bootleg mapper support. Best wishes!!!! :P

**Comment by Redcrafter** [comment]
Hey Robin
Thanks for notifying me on this.
I know exactly what this is. I'm compiling using msvc and it supports this even though it's not in the c++ standard. Also yeah i accidentally pushed the Gameboy include.

The build should be fixed with b5655363844d3715150f20450c98ee0bb3a5f915

The Gameboy implementation is going well. Should be ready soon.
